src/Simulation/TargetDefinitions/Intrinsic/Z.qs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Intrinsic {
+    /// # Summary
+    /// Applies the Pauli $Z$ gate.
+    ///
+    /// # Description
+    /// \begin{align}
+    ///     \sigma_z \mathrel{:=}
+    ///     \begin{bmatrix}
+    ///         1 & 0 \\\\
+    ///         0 & -1
+    ///     \end{bmatrix}.
+    /// \end{align}
+    ///
+    /// # Input
+    /// ## qubit
+    /// Qubit to which the gate should be applied.
+    operation Z (qubit : Qubit) : Unit is Adj + Ctl {
+        body intrinsic;
```

Comment:
Does the TargetDefinitions folder contain the particular intrinsics and decompositions for the QuantumSimulator? Or is it more general than that?

Reply:
It's more general than that, and relates to the other question you had about separating callables into individual files. See my answer there.